DPG/Tasks/AOTTrack/tagAndProbeDmesons.cxx

```diff
@@ -233,18 +262,103 @@ struct TagTwoProngDisplacedVertices {
+                   const uint8_t channel,
+                   std::vector<int>& pdgDecayMothers)
+  {
+    auto pdgTagMother = 0;
+    int pdgProbeParticle = -1;
+
+    if (channel == aod::tagandprobe::TagChannels::DplusToKPiPi) {
+      pdgTagMother = constants::physics::Pdg::kDPlus;
+      pdgProbeParticle = 321; // Ka
+    } else if (channel == aod::tagandprobe::TagChannels::DsOrDplusToKKPi) {
+      pdgTagMother = constants::physics::Pdg::kPhi;
+      pdgProbeParticle = 211; // Pi
+    } else {
+      pdgTagMother = constants::physics::Pdg::kD0;
+      pdgProbeParticle = 211; // Pi
+    }
+
+    if (!firsTrack.has_mcParticle() || !secondTrack.has_mcParticle()) {
+      return -999;
+    } else {
+      auto firstMcTrack = firsTrack.template mcParticle_as<aod::McParticles>();
+      auto secondMcTrack = secondTrack.template mcParticle_as<aod::McParticles>();
+      auto firstTrackMotherId = RecoDecay::getMother(particlesMc, firstMcTrack, pdgTagMother, true);
+      auto secondTrackMotherId = RecoDecay::getMother(particlesMc, secondMcTrack, pdgTagMother, true);
+
+      if ((firstTrackMotherId == secondTrackMotherId) && (firstTrackMotherId != -1)) {
+        auto particleMother = particlesMc.rawIteratorAt(firstTrackMotherId);
+
+        /// π±π± for D± → K∓π±π± decays
+        if (channel == aod::tagandprobe::TagChannels::DplusToKPiPi) {
+          auto particleMother = particlesMc.rawIteratorAt(firstTrackMotherId);
+          auto daughters = particleMother.template daughters_as<aod::McParticles>();
+          // Check if the probe is within the mother's particle daughters
+          for (auto& daughter : daughters) {
+            if (std::abs(daughter.pdgCode()) == pdgProbeParticle) {
+              return 1;
+            }
+          }
+        }
+
+        ///  K∓K± for φ from Ds± or D± → φπ± decays
+        ///  K∓π± for D0 from D±* → D0π± decays
+        else {
+          for (auto pdgGrandMother : pdgDecayMothers) {
+            auto grandMotherId = RecoDecay::getMother(particlesMc, particleMother, pdgGrandMother, true);
+            auto particleGrandMother = particlesMc.rawIteratorAt(grandMotherId);
+            if (grandMotherId != -1) {
+              auto daughters = particleGrandMother.template daughters_as<aod::McParticles>();
+              // Check if the probe is within the GrandMother's particle daughters
+              for (auto& daughter : daughters) {
+                if (std::abs(daughter.pdgCode()) == pdgProbeParticle) {
+                  return 1;
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    return 0;
+  }
+
   template <typename Pvec>
   bool isSelectedInvariantMass(const Pvec& pVecTrackFirst,
                                const Pvec& pVecTrackSecond,
                                const uint8_t channel,
-                               float& invMass2)
+                               float& invMass2,
+                               const int& ptBin)
```

Comment:
why did you add this? where do you use this `ptBin` in the function?

Reply:
In line 533 and 599 we calculate the ptBin once for each function "computeCombinatorialSameCharge" and "computeCombinatorialOppositeCharge". Then we pass this information to the functions which need it. Particularly, in this case we use this variable to get the topological cuts of interest.
Thank you.